src/lib/product/ProductImport.ts

```diff
@@ -96,10 +96,12 @@ export class ProductImport {
             let allProducts = [];
             this.products.forEach(product => { allProducts = [...allProducts, ... product.getProducts()] });
             const allProductsRTfix = Util.fixRecordTypes(allProducts, recordTypes, 'Product2');
-            const allProductsWithouthRelationships = Util.sanitizeDeepForUpsert(allProductsRTfix);
+            const allProductsWithoutRelationships = Util.sanitizeDeepForUpsert(allProductsRTfix);
 
-            await Upsert.upsertData(this.connection, Util.sanitizeForUpsert(allProductsWithouthRelationships), 'Product2');
-            await Upsert.upsertData(this.connection, Util.sanitizeForUpsert(allProductsRTfix), 'Product2');
+            await Upsert.upsertData(this.connection, Util.sanitizeForUpsert(allProductsWithoutRelationships), 'Product2');
+            //05.08.2020 SZILN - ECPQ-4615 - As the data are now upserted again in case of failure, 
+            //the second upsert of not-deeply-sanitized data is commented out.
+            //await Upsert.upsertData(this.connection, Util.sanitizeForUpsert(allProductsRTfix), 'Product2');
```

Comment:
@Epasper nie wiem czy do końca ogarniam tą zmiankę, ale chyba warto sprawdzić czy się rozumiemy. 

2-krokowy upsert produktów miał na celu poprawne przeniesienie relacji. Przykład: masz produkt który ma pod sobą opcję (opcja ma lookup do produktu). Jeśli wrzucisz opcję od razu to rzuci Ci error że nie ma parenta. Dlatego w pierwszym upsercie były wrzucane produkty i opcje bez tych relacji - drugi upsert miał na celu przeniesienie tych relacji.

Reply:
Jasne, źle zrozumiałem kod - myślałem że sanitizeDeepForUpsert czyści niepotrzebne relacje, ale jak teraz czytam co właściwie robi, to widzę, że faktycznie czyści wszystkie relacje. Cofnąłem moją zmianę.